### ISSUE TITLE
plt.show( ) not working (can't get figures to display in external window) when using jupyter QTconsole

### ISSUE THREAD

**Issue by beatbox13** [matplotlib/matplotlib#12397]
### Bug report

**Bug summary**

I am trying to have my figures open up in external windows using the plt.show() function. However, It gives me no output other than something such as:
```<Figure size 640x480 with 1 Axes>```

I am running linux ubuntu latest release and matplotlib 2.2.2, 
qtconsole 4.3.1, 
anaconda client 1.6.14 
anaconda navigator 1.8.7 
anaconda project 0.8.2, 
jupyter 1.0.0    
jupyter-client 5.2.3    
jupyter-console 5.2.0    
jupyter-core 4.4.0    
jupyterlab 0.32.1   
jupyterlab-launcher 0.10.5 
python 3.6.5

```python

import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
fig= plt.figure()
ax= fig.add_subplot(111)
ax.plot([1, 2, 3, 4], [10, 20, 25, 30], color= "lightblue", linewidth= 3)
ax.scatter([0.3, 3.8, 1.2, 2.5], [11, 25, 9, 26], color= "darkgreen", marker= "^")
ax.set_xlim(0.5, 4.5)
plt.show()

```

**Actual outcome**

```
<Figure size 640x480 with 1 Axes>
```

**Expected outcome**

figure open in external window

**Matplotlib version**

  * Operating system:
  * Matplotlib version: 
  * Matplotlib backend (`print(matplotlib.get_backend())`):
  * Python version:
  * Jupyter version (if applicable):
  * Other libraries: 

Installed through anaconda

**Comment by ImportanceOfBeingErnest** [comment]
Qt console, as many other interactive interpreters based on IPython, will require a backend being set. You will find ways to do that [all over the internet](https://www.google.de/search?q=qtconsole+matplotlib+show+figure+site:stackoverflow.com). In this case I suppose you want to add `%matplotlib qt` before starting to plot.  
If you need further help with that, asking a question on stackoverflow, or via the [mailing list](https://mail.python.org/mailman/listinfo/matplotlib-users) are two options you have. 

Closing because there seems no issue with matplotlib involved here.

**Comment by cpoptic** [comment]
Make sure you're running `%matplotlibe inline` or `%matplotlibe notebook`  *before* importing the matplotlib library.  Otherwise you'll just see a blank   `<Figure size 640x480 with 1 Axes>`

**Comment by arthurmloureiro** [comment]
Hi, I have the same problem. I have tried all the suggestions online and for some reason, I cannot produce plots inline in any way. I have tried all backends possible. It was a recent update that broke this. Current verion is 2.2.9

**Comment by nesergen** [comment]
Same problem. Advices didn't help. JupyterLab Version 2.2.9

**Comment by jklymak** [comment]
@nsergen, please open a new issue with exactly what you are trying to do, and what did not help.  Please also search the jupyter help.
